app/helpers/podcasts_helper.rb

```diff
@@ -101,4 +101,54 @@ def disable_non_compliant_language(language)
       language
     end
   end
+
+  def subscribe_link_platform_label(link)
+    I18n.t("helpers.label.podcast.subscribe_link.#{link.platform}")
+  end
+
+  def subscribe_link_id_label(link)
+    if link.uses_apple_id?
+      I18n.t("helpers.label.podcast.subscribe_link.id.apple")
+    elsif link.uses_unique_id?
+      I18n.t("helpers.label.podcast.subscribe_link.id.unique", platform: subscribe_link_platform_label(link))
+    elsif link.uses_feed_url?
+      I18n.t("helpers.label.podcast.subscribe_link.id.feed")
+    elsif link.uses_podcast_guid?
+      I18n.t("helpers.label.podcast.subscribe_link.id.guid")
+    elsif link.uses_pod_index_id?
+      I18n.t("helpers.label.podcast.subscribe_link.id.pod_index")
+    end
+  end
+
+  def subscribe_link_help(link)
+    if link.uses_apple_id?
+      I18n.t("podcast_engagement.form_subscribe_links.help.apple")
+    elsif link.uses_unique_id?
+      I18n.t("podcast_engagement.form_subscribe_links.help.unique")
+    elsif link.uses_feed_url?
+      I18n.t("podcast_engagement.form_subscribe_links.help.feed")
+    elsif link.uses_podcast_guid?
+      I18n.t("podcast_engagement.form_subscribe_links.help.guid")
+    elsif link.uses_pod_index_id?
+      I18n.t("podcast_engagement.form_subscribe_links.help.pod_index")
+    end
+  end
+
+  def subscribe_link_options(podcast)
+    selected = podcast.subscribe_links.select(&:persisted?).map { |sl| sl.platform }
+
+    SubscribeLink::PLATFORMS.filter { |p| !selected.include?(p) }.map { |p| {platform: p, icon: SubscribeLink::PLATFORM_ICONS[p]} }
+  end
+
+  def subscribe_link_external_id(podcast, platform)
```

Comment:
this is the helper that pre-populates the field. apple id gets pre-filled only if there is already an apple id to pull from. feed url uses the `public_url` and guid uses `guid`, but if these need to use different attributes let me know.